src/Build/BackEnd/Components/Logging/EventSourceSink.cs

```diff
@@ -217,66 +217,63 @@ public void Consume(BuildEventArgs buildEvent, int sinkId)
         /// </summary>
         public void Consume(BuildEventArgs buildEvent)
         {
-            // FXCop may complain that there are unecessary casts here, and there are, but
-            // using "as" and allocating another variable for each event is extremely costly
-            // and is much slower then this approach even with the additional casts
-            if (buildEvent is BuildMessageEventArgs)
+            if (buildEvent is BuildMessageEventArgs buildMessageEvent)
```

Comment:
Nit:
Maybe nicer as a switch statement?

Reply:
I completely forgot about this feature. Done, thank you! Too bad switch expressions cannot be used here, they would make it even nicer.